include/odp/api/spec/ipsec.h

```diff
@@ -238,6 +238,11 @@ typedef struct odp_ipsec_capability_t {
 	 */
 	odp_support_t retain_header;
 
+	/**
+	 * Inner packet checksum check offload support in inbound direction.
+	 */
+	odp_proto_chksums_t chksums_in;
```

Comment:
Is there no need for a corresponding `chksums_out` capability?

Reply:
There is one indeed.

Reply:
Why? The rationale goes that if checksumming is requested in the outbound direction the implementation can always calculate in in SW since that is what the application would have to do otherwise. Inbound direction is different since the need for L4 checksum checking (i.e. is the packet destined to this system of just forwarded) is not yet known at the time of reception (so if an implementation sets the inbound capability, it should mean that it can do checksumming clearly more efficiently than pure SW).

Reply:
OK

Reply:
About 2 vs. 3 step capability. Two level should be enough (no, yes). If there are 3 levels (no, yes in SW, yes in HW), throughput would be optimized is application enabled only when done in HW. Application would not request SW checksum for all packets, as it can filter out those that actually need it (e.g. half of the packets) and then do it in SW only for those.